### PR TITLE
fix(unittest): Init with `dbfilename=` before attempting to `save`

### PR DESCRIPTION
**Pull request by chakaz**
This is a pretty recent regression.

### CODE REVIEW

**Comment by romange** [comment]
What has changed? Why do we need to reset the service now with `dbfilename=...`?

**Comment by chakaz** [comment]
> What has changed? Why do we need to reset the service now with `dbfilename=...`?

A change by some @romange :rofl: 

https://github.com/dragonflydb/dragonfly/commit/8a65aec805c37d55395fab75fed41c258f1f5839

File: src/server/rdb_test.cc [hunk 1]
```diff
@@ -33,33 +32,18 @@ using absl::StrCat;
...
-  // Disable save on shutdown
-  SetFlag(&FLAGS_dbfilename, "");
-
-  auto rdb_files = io::StatFiles("rdbtestdump*");
-  CHECK(rdb_files);
-  for (const auto& fl : *rdb_files) {
-    unlink(fl.name.c_str());
-  }
-  BaseFamilyTest::TearDown();
+  InitWithDbFilename();
```
Comment (romange) [inline src/server/rdb_test.cc:46]: Ok, I understand now the context but I think for these tests `SetFlag(&FLAGS_dbfilename, "rdbtestdump");` is enough.

File: src/server/test_utils.h [hunk 2]
```diff
@@ -94,6 +94,9 @@ class BaseFamilyTest : public ::testing::Test {
 
   void ShutdownService();
 
+  void InitWithDbFilename();
+  void DelSerializedFiles();
```
Comment (romange) [inline src/server/test_utils.h:98]: nit: maybe rename it to `CleanupSnapshots()` ?

**Review by romange — DISMISSED** [review]

File: src/server/rdb_test.cc (hunk 1 above)
Comment (chakaz) [reply src/server/rdb_test.cc:46]: Even if not, that's a good infra to have, which already semi exists for `rdb_test`, I think it's better to keep it for future use as well

**Review by romange — APPROVED** [review]
I wish GTest would have the way to provide per test, presetup customizations. Unfortunately they do not have it, so we need to restart everything within the test. One way to solve this would be to implement a `SaveTest` class that derives from `BaseFamilyTest`

**Comment by chakaz** [comment]
That would be a useful feature indeed...
But it's not too bad, we have a few functions that reset the service anyway

**Pull request merged**